gr-analog/python/analog/qa_fastnoise.py

```diff
@@ -23,67 +23,117 @@
-            sink = blocks.vector_sink_f()
-            paths[src] = (src, head, sink)
+        # mean, variance
+        self.assertAlmostEqual(data.mean(), 0, places=2)
+        self.assertAlmostEqual(data.var(), (1-(-1))**2./12, places=3)
 
-        for src in cplx_srcs:
-            head = blocks.head(nitems=NUM_ITEMS, sizeof_stream_item=gr.sizeof_gr_complex)
-            sink = blocks.vector_sink_c()
-            paths[src] = (src, head, sink)
-        for path in paths.itervalues():
-            tb.connect(*path)
+    def test_001_real_gaussian_moments(self):
+        data = self.run_test_real(analog.GR_GAUSSIAN)
 
-        tb.run()
-        self.data = {src: path[2].data() for src, path in paths.iteritems()}
-        data = self.data
-        # min, max
-        self.assertAlmostEqual(min(data[self.uni_real_src]), -1, places=4)
-        self.assertAlmostEqual(max(data[self.uni_real_src]), 1, places=4)
         # mean, variance
-        data = numpy.array(data[self.uni_real_src])
-        self.assertAlmostEqual(data.mean(), 0, places = 2)
-        self.assertAlmostEqual(data.var(), (1-(-1))**2./12, places = 3)
-        data = numpy.array(self.data[self.norm_real_src])
         self.assertAlmostEqual(data.mean(), 0, places=2)
         self.assertAlmostEqual(data.var(), 1, places=2)
-        data = numpy.array(self.data[self.lapl_real_src])
+
+    def test_001_real_laplacian_moments(self):
+        data = self.run_test_real(analog.GR_LAPLACIAN)
+
+        # mean, variance
         self.assertAlmostEqual(data.mean(), 0, places=2)
         self.assertAlmostEqual(data.var(), 2, places=2)
 
+    def test_001_complex_uniform_moments(self):
+        data = self.run_test_complex(analog.GR_UNIFORM)
+
+        # mean, variance
+        self.assertAlmostEqual(data.real.mean(), 0, places=2)
+        self.assertAlmostEqual(data.real.var(), 0.5*(1-(-1))**2./12, places=3)
+
+        self.assertAlmostEqual(data.imag.mean(), 0, places=2)
+        self.assertAlmostEqual(data.imag.var(), 0.5*(1-(-1))**2./12, places=3)
+
+    def test_001_complex_gaussian_moments(self):
+        data = self.run_test_complex(analog.GR_GAUSSIAN)
+
+        # mean, variance
+        self.assertAlmostEqual(data.real.mean(), 0, places=2)
+        self.assertAlmostEqual(data.real.var(), 0.5, places=2)
+
+        self.assertAlmostEqual(data.imag.mean(), 0, places=2)
+        self.assertAlmostEqual(data.imag.var(), 0.5, places=2)
+
+    def test_002_real_uniform_reproducibility(self):
+        data1 = self.run_test_real(analog.GR_UNIFORM)
+        data2 = self.run_test_real(analog.GR_UNIFORM)
+
+        # It's pseudoramdo thus must be equal
+        self.assertTrue(numpy.array_equal(data1, data2))
+
+    def test_002_real_gaussian_reproducibility(self):
+        data1 = self.run_test_real(analog.GR_GAUSSIAN)
+        data2 = self.run_test_real(analog.GR_GAUSSIAN)
+
+        self.assertTrue(numpy.array_equal(data1, data2))
+
+    def test_003_real_uniform_pool(self):
```

Comment:
Did you autogen this section? The formatting looks different from before.

Reply:
No autogen on my part, don't know about the formatting, Did I format the full file by accident?

Reply:
at this point, since the original maint test of pre-my-kind-of-fix is nowhere to be found in this test anymore, I'd be OK with that.